docs/source/daml/patterns/daml/MultiplePartyAgreement.daml

```diff
@@ -11,60 +11,82 @@ import Utilities
 import DA.List
 
 -- start snippet: agreement template
-template Agreement
+template ContractPlaceholder
   with
     signatories: [Party]
   where
     signatory signatories
+    ensure
+      unique signatories
+  -- The rest of the template to be agreed to would follow here
 -- end snippet: agreement template
 
 -- start snippet: first half pending template
 template Pending
   with
-    agree: Agreement
-    signed: [Party]
-    toSign: [Party]
+    finalContract: ContractPlaceholder
+    alreadySigned: [Party]
   where
-    signatory signed
-    observer toSign
+    signatory alreadySigned
+    observer finalContract.signatories
+    ensure
+      -- Can't have duplicate signatories
+      unique alreadySigned
+
+    -- The parties who need to sign is the finalContract.signatories with alreadySigned filtered out
+    let toSign = filter (`notElem` alreadySigned) finalContract.signatories
 
     choice Sign : ContractId Pending with
```

Comment:
Flexible controller choice is a nice change.

Reply:
I agree!